enum34 is only required for Python < 3.4, and is not compatible at all with Python 3.6+
see also https://github.com/tensorflow/tensorflow/pull/14418
author: Kenneth Host (HPC-UGent)
--- tensorflow/tools/pip_package/setup.py.orig	2017-11-09 13:57:14.841957214 +0100
+++ tensorflow/tools/pip_package/setup.py	2017-11-09 14:01:16.744310560 +0100
@@ -32,7 +32,6 @@
 _VERSION = '1.4.0'
 
 REQUIRED_PACKAGES = [
-    'enum34 >= 1.1.6',
     'numpy >= 1.12.1',
     'six >= 1.10.0',
     'protobuf >= 3.3.0',
@@ -46,6 +45,11 @@
   sys.argv.remove('--project_name')
   sys.argv.pop(project_name_idx)
 
+# enum34 is only required for Python 3.4 or older
+# see https://bitbucket.org/stoneleaf/enum34/issues/19
+if not (sys.version_info.major >= 3 and sys.version_info.minor >= 4):
+    REQUIRED_PACKAGES.append('enum34 >= 1.1.6')
+
 # python3 requires wheel 0.26
 if sys.version_info.major == 3:
   REQUIRED_PACKAGES.append('wheel >= 0.26')
